crates/typos-cli/src/bin/typos-cli/report.rs

```diff
@@ -280,6 +286,148 @@ impl Report for PrintJson {
     }
 }
 
+#[derive(Debug)]
+pub(crate) struct PrintSarif {
+    results: Mutex<RefCell<Vec<sarif::Result>>>,
+    error: Mutex<RefCell<Vec<String>>>,
+}
+
+impl Default for PrintSarif {
+    fn default() -> Self {
+        Self {
+            results: Mutex::new(RefCell::new(Vec::new())),
+            error: Mutex::new(RefCell::new(Vec::new())),
+        }
+    }
+}
+
+fn error_to_io_error(error: Box<dyn std::error::Error>) -> std::io::Error {
```

Comment:
You can skip the `e.into()`s by making this
```rust
fn error_to_io_error(error: impl Into<Box<dyn std::error::Error>>) -> std::io::Error {
    let error = error.into();
    ...
}
```
Though if all you are doing is `to_string`, you can instead do
```rust
fn error_to_io_error(error: impl std::fmt::Display) -> std::io::Error {
    ...
}
```